django/db/backends/mysql/operations.py

```diff
@@ -143,6 +143,14 @@ def time_trunc_sql(self, lookup_type, field_name):
     def date_interval_sql(self, timedelta):
         return 'INTERVAL %s MICROSECOND' % duration_microseconds(timedelta)
 
+    def fetch_returned_insert_rows(self, cursor):
+        """
+        Given a cursor object that has just performed an INSERT...RETURNING
+        statement into a table, return the tuple of returned data.
+        """
+        # Convert to list to communicate this is for multiple rows
+        return list(cursor.fetchall())
```

Comment:
I wonder why it works without `list()` on PostgreSQL :thinking:

Reply:
psycopg2 returns lists I think

This is for this check: https://github.com/django/django/blob/8f2a6c76d19e4010c4683b20ed7f1eb4b07c17eb/django/db/models/query.py#L1257

Not sure if it could be changed.

Reply:
Due to the [PEP-249](https://www.python.org/dev/peps/pep-0249/#fetchall) it can be any sequence, so we can change this in advance.

Reply:
... or simply accept `tuple` if that's the case on MariaDB.

Reply:
I need to dig around a bit more to change that then

Reply:
I checked and on PostgreSQL we have a list of tuples for multiple rows and a tuple for a single row; on MariaDB we have a tuple of tuples for multiple rows and a tuple for a single row :confused: 
It's a really implicit logic, we should support both formats in:
https://github.com/django/django/blob/8f2a6c76d19e4010c4683b20ed7f1eb4b07c17eb/django/db/models/query.py#L1257-L1260

Reply:
I've pushed a commit that changes this but it's not quite working, will come back to it